airbyte-scheduler/persistence/src/test/java/io/airbyte/scheduler/persistence/job_tracker/JobTrackerTest.java

```diff
@@ -320,19 +331,57 @@ void testConfigToMetadata() throws IOException {
   }
 
   void testAsynchronousAttempt(final ConfigType configType) throws ConfigNotFoundException, IOException, JsonValidationException {
-    testAsynchronousAttempt(configType, Collections.emptyMap());
+    testAsynchronousAttempt(configType, getJobWithAttemptsMock(configType, LONG_JOB_ID), Collections.emptyMap());
   }
 
   void testAsynchronousAttempt(final ConfigType configType, final Map<String, Object> additionalExpectedMetadata)
       throws ConfigNotFoundException, IOException, JsonValidationException {
-    // for sync the job id is a long not a uuid.
-    final long jobId = 10L;
+    testAsynchronousAttempt(configType, getJobWithAttemptsMock(configType, LONG_JOB_ID), additionalExpectedMetadata);
+  }
 
-    final ImmutableMap<String, Object> metadata = getJobMetadata(configType, jobId);
-    final Job job = getJobWithAttemptsMock(configType, jobId);
+  void testAsynchronousAttemptWithFailures(final ConfigType configType, final Map<String, Object> additionalExpectedMetadata)
+      throws ConfigNotFoundException, IOException, JsonValidationException {
+    final JsonNode configFailureJson = Jsons.jsonNode(ImmutableMap.of(
+        "failureOrigin", "source",
+        "failureType", "configError",
+        "internalMessage", "Internal config error error msg",
+        "externalMessage", "Config error related msg",
+        "metadata", ImmutableMap.of("some", "metadata"),
+        "retryable", true,
+        "timestamp", 1010));
+
+    final JsonNode systemFailureJson = Jsons.jsonNode(ImmutableMap.of(
+        "failureOrigin", "replication",
+        "failureType", "systemError",
+        "internalMessage", "Internal system error error msg",
+        "externalMessage", "System error related msg",
+        "metadata", ImmutableMap.of("some", "metadata"),
+        "retryable", true,
+        "timestamp", 1100));
+
+    final JsonNode unknownFailureJson = Jsons.jsonNode(ImmutableMap.of(
+        "failureOrigin", "unknown",
+        "failureType", "unknown",
```

Comment:
For this case, we should test null failureOrigin and failureType instead of `unknown`, see https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/FailureReason.yaml#L14

I'm currently writing a migration to remove these enum values and update any records with `unknown` values to `null`. Basically, because these are optional fields, it's pretty confusing to allow both 'null' and 'unknown' values since they end up meaning the same thing.

Reply:
ah ok, in this case I guess the json can explicitly include the failureOrigin and failureType keys with value `null`.  Or do you think we just shouldn't include the keys at all in this case?  I guess either can work depending on how the data is transformed after being loaded into bigquery.

Reply:
Good question, I think including them with the value 'null' is the most explicit, so I'd probably go with that and we'll revisit if we find out that makes things harder in BigQuery for some reason

Reply:
cool, sounds good